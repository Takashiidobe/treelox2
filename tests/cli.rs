use std::process::Output;

use assert_cmd::cargo::CommandCargoExt;
use insta::{assert_yaml_snapshot, glob};
use insta_cmd::Command;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct TestOutput {
    status: i32,
    stdout: Vec<String>,
    stderr: Vec<String>,
}

#[test]
fn reference_files() {
    glob!("../test-files", "**/*.lox", |path| {
        let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();

        let Output {
            status,
            stdout,
            stderr,
        } = cmd.arg(path).output().unwrap();

        let test_output = TestOutput {
            status: status.code().unwrap(),
            stdout: String::from_utf8_lossy(&stdout)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
            stderr: String::from_utf8_lossy(&stderr)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
        };

        assert_yaml_snapshot!(test_output);
    });
}
