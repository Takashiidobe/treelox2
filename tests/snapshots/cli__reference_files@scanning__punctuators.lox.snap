---
source: tests/cli.rs
expression: test_output
input_file: test-files/scanning/punctuators.lox
---
status: 101
stdout: []
stderr:
  - "[line 1] Error at ')': Expect expression."
  - "thread 'main' panicked at src/parser.rs:142:9:"
  - "Couldn't parse Token { type: RightParen, literal: None, loc: Location { lexeme: \")\", line: 1 } }"
  - "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace"
